react-components/src/data-providers/core-dm-provider/cadAndPointCloudAssetQuery.ts

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+import { type QueryRequest } from '@cognite/sdk';
+import { type Source, type InstanceFilter } from '../FdmSDK';
+import { cogniteAssetSourceWithProperties } from './cogniteAssetSourceWithProperties';
+import {
+  COGNITE_CAD_NODE_SOURCE,
+  COGNITE_ASSET_SOURCE,
+  COGNITE_POINT_CLOUD_VOLUME_SOURCE,
+  CORE_DM_3D_CONTAINER_SPACE
+} from './dataModels';
+
+export function cadAndPointCloudAssetQuery(
+  sourcesToSearch: Source[],
+  filter: InstanceFilter | undefined,
+  limit: number
+): QueryRequest {
+  // This return value throws away useful type information about the result, but I haven't found a way to
+  // make the full type easily expressible
```

Comment:
Oh, haha :sweat_smile: I just reached out to Christopher about it. It's very tempting to just make eslint ignore this function, so that we can return the full query type